compress-mpq.c

```diff
@@ -557,97 +552,77 @@ void InitCache() {
     }
 }
 
-void CachePacked(char *path, size_t insize, size_t outsize, uint32_t flags, unsigned char *content, unsigned char *out){
-    // Sanitize the input path by replacing slashes with hyphens.
-    char sanitized_path[1024];
-    SanitizePath(path, sanitized_path);
+void CachePacked(const char *path, const unsigned char *out, const size_t outsize, const unsigned char *content, const size_t insize){
+    char cache_path[1024];
 
-    // Construct the full file path: "{CWD}/cache/{sanitized_path}"
-    char full_path[1024];
-    snprintf(full_path, sizeof(full_path), "./cache/%s", sanitized_path);
+    {
+      char in_archive_path64[971];
+      char content_hash64[45];
+      char content_hash[32];
+
+      // Construct the full file path: "{CWD}/cache/{content_hash64}-{in_archive_path64}"
```

Comment:
What's the reason to use in_archive_path? You would get more cache hits if just using the content hash.

Reply:
I think it's neat to be able to inspect the contents of a cache. That was also part of the motivation for the original design that stored the whole files. To further this goal, I'd rather use a readable restricted-charset encoding, such as [DashyCode](https://github.com/smogon/pokemon-showdown/blob/master/lib/DASHYCODE.md), but I am not aware of any non-JS ports, or C alternatives.

Regarding cache hits, if ``in_archive_path`` doesn't match, they would be extremely likely to be false positives, though! Unless the use case is for any obfuscators out there that generate random import names? Is that something that exists and/or you'd want to support?